src/dotnet-install.sh

```diff
@@ -1211,15 +1210,10 @@ generate_akams_links() {
         effective_versions+=($effective_version)
         link_types+=("aka.ms")
 
-        if [[ "$dry_run" == true ]]; then
-            print_dry_run "$download_link" "" "$effective_version"
-            return 1
-        fi
-
         #  Check if the SDK version is already installed.
-        if is_dotnet_package_installed "$install_root" "$asset_relative_path" "$effective_version"; then
+        if [[ "$dry_run" != true ]] && is_dotnet_package_installed "$install_root" "$asset_relative_path" "$effective_version"; then
```

Comment:
is it reasonable to lift the restriction on dry run and also check if version is installed in dry run mode?

Reply:
Potentially, yes. But since there is no ask for this from any of our customers, I prefer to skip it.
Lifting the restriction doesn't just work. We should prevent the app from stopping after the detection of installation and store this information to be used later when printing dry run.